addons/website_sale/models/sale_order.py

```diff
@@ -287,7 +287,7 @@ def _cart_update(self, product_id, line_id=None, add_qty=0, set_qty=0, **kwargs)
         if add_qty and (not product or not product._is_add_to_cart_allowed()):
             raise UserError(_("The given product does not exist therefore it cannot be added to cart."))
 
-        if line_id is not False:
```

Comment:
Unfortunately, doing this would break other flows. The correct place to fix this would be to remove this line: https://github.com/odoo/odoo/blob/97a64f75c454baa969de7bbe33550a0adaca463b/addons/website_sale/controllers/combo_configurator.py#L74
However, that's not enough. Indeed, if you added the same combo twice, we'd always consider them identical, even if they were configured differently. So we'd need to add extra logic to check whether all combo item lines are identical.